src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityCrate.java

```diff
@@ -148,8 +148,19 @@ public ModularPanel buildUI(PosGuiData guiData, PanelSyncManager guiSyncManager)
         for (int i = 0; i < rows; i++) {
             widgets.add(new ArrayList<>());
             for (int j = 0; j < this.rowSize; j++) {
-                widgets.get(i).add(new ItemSlot().slot(SyncHandlers.itemSlot(inventory, i * rowSize + j)
-                        .slotGroup("item_inv")));
+                int index = i * rowSize + j;
+                widgets.get(i).add(new ItemSlot().slot(SyncHandlers.itemSlot(inventory, index)
+                        .slotGroup("item_inv")
+                        .changeListener((newItem, onlyAmountChanged, client, init) -> {
+                            if (!onlyAmountChanged && !client && !init) {
```

Comment:
Why do we check `onlyAmountChanged` here? I would imagine that we would want to update neighbors if anything about the contained stacks changes, including the stacksize.